tests/test_paper.py

```diff
@@ -0,0 +1,209 @@
+'''
+Paper-mode testing
+'''
+
+import trio
+from exceptiongroup import BaseExceptionGroup
+from typing import (
+    AsyncContextManager,
+    Literal,
+)
+
+import pytest
+import tractor
+from tractor._exceptions import ContextCancelled
+from uuid import uuid4
+from functools import partial
+
+from piker.log import get_logger
+from piker.clearing._messages import Order
+from piker.pp import (
+    open_trade_ledger,
+    open_pps,
+)
+
+log = get_logger(__name__)
+
+
+def get_fqsn(broker, symbol):
+    fqsn = f'{symbol}.{broker}'
+    return (fqsn, symbol, broker)
+
+
+oid = ''
+test_exec_mode = 'live'
+(fqsn, symbol, broker) = get_fqsn('kraken', 'xbtusdt')
+brokers = [broker]
+account = 'paper'
+
+
+async def _async_main(
+    open_test_pikerd_and_ems: AsyncContextManager,
+    action: Literal['buy', 'sell'] | None = None,
+    price: int = 30000,
+    executions: int = 1,
+    size: float = 0.01,
+    # Assert options
+    assert_entries: bool = False,
+    assert_pps: bool = False,
+    assert_zeroed_pps: bool = False,
+    assert_msg: bool = False,
+   ) -> None:
+    '''
+    Start piker, place a trade and assert data in 
```

Comment:
there's still trailing whitespace on these next 2 lines when i open in an editor. you can't see it here because there's no diff with previous state 😉